### PR TITLE
3. Primary section

### PR DESCRIPTION
**Pull request by verenichyi**
- Add header in primary section
- Change file structure, put route paths into enum
- Add go back button in case page not found
- Change header appearance if page is not primary
- Add search form
- Change header icons format to svg
- Show selected nav link
- use `Context` API for primary page indication

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (echonok) [inline src/App.tsx:1]: please sort imports like
- framework core-libs
- external libs/packages
- components

File: src/modules/PrimarySection/PrimarySection.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    <section className={styles.primarySection}>
+      <div className={styles.backgroundContainer}>
+        <Header />
+        <Container>
+          <div className={styles.content}>
+            <h1 className={styles.title}>Book pool time and amenities at luxury hotels</h1>
+            <form action="#" className="header__search-form search-form">
+              <div className="search-from__filter">
+                <ul className="search-from__filter-list">
+                  <li className="search-from__filter-item search-from__filter-item_active">All</li>
```
Comment (echonok) [inline src/modules/PrimarySection/PrimarySection.tsx]: please reside it to an array and loop through it

File: src/modules/PrimarySection/PrimarySection.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import Header from '../Header';
+
+const PrimarySection = () => {
+  return (
+    <section className={styles.primarySection}>
+      <div className={styles.backgroundContainer}>
+        <Header />
+        <Container>
+          <div className={styles.content}>
+            <h1 className={styles.title}>Book pool time and amenities at luxury hotels</h1>
```
Comment (echonok) [inline src/modules/PrimarySection/PrimarySection.tsx]: please, reside it to a variable. We may want to change it dynamically

**Review by echonok — CHANGES_REQUESTED** [review]

File: src/modules/PrimarySection/PrimarySection.tsx (hunk 2 above)
Comment (verenichyi) [reply src/modules/PrimarySection/PrimarySection.tsx]: Yeah, I was going to do it exactly like this. I just haven't touched it yet, just did copy/paste from initial.

File: src/components/NavMenu/NavMenu.tsx [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
 
 const navLinks = [
   { to: 'help', link: 'Help' },
   { to: 'how-it-works', link: 'How it works' },
   { to: 'for-partners', link: 'For partners' },
 ];
 
-const NavMenu = () => {
+interface Props {
+  isPrimaryPage?: boolean;
```
Comment (echonok) [inline src/components/NavMenu/NavMenu.tsx]: is it necessary to use optional here?

File: src/components/NavMenu/styles.module.scss [hunk 5]
```diff
@@ -58,7 +57,7 @@
     right: -100%;
     width: 100%;
     height: 100%;
-    background-color: #2EB7EEFF;
+    background-color: #2eb7eeff;
```
Comment (echonok) [inline src/components/NavMenu/styles.module.scss]: maybe move to a variable?

File: src/components/SearchSelect/SearchSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import styles from './styles.module.scss';
+import locationIcon from '../../assets/images/header/location.png';
+
+const SearchSelect = () => {
+  return (
+    <div className={styles.select}>
+      <div>Los Angeles, CA</div>
```
Comment (echonok) [inline src/components/SearchSelect/SearchSelect.tsx]: Moved to a variable

**Review by echonok — CHANGES_REQUESTED** [review]

**Review by echonok — APPROVED** [review]

**Pull request merged**